app/components/license/CurrentVersion.tsx

```diff
@@ -1,33 +1,54 @@
-// @ts-nocheck File not migrated fully to TS
+import type { SemanticICONS } from 'semantic-ui-react';
 import i18n from 'i18next';
-import _ from 'lodash';
-import PropTypes from 'prop-types';
+import { join, compact, capitalize, startCase, isEmpty, map } from 'lodash';
 import React from 'react';
 
 import { Icon, Header, Message, Segment, Table } from '../basic';
+import type { VersionResponse } from '../../../backend/handler/AuthHandler.types';
 
-export default function CurrentVersion({ version = {} }) {
-    version.distro = _.join([version.distribution, version.distro_release], ' ');
-    version.full_version = `${_.join(_.compact([version.version, version.build, version.date, version.commit]), ' ')}
-         (${_.capitalize(version.edition)})`;
+interface ExtendedVersion extends Partial<VersionResponse> {
+    distro: string;
+    // eslint-disable-next-line camelcase
+    full_version: string;
+}
+
+interface CurrentVersionProps {
+    version: Partial<VersionResponse>;
+}
 
-    const fields = [
+interface Field {
```

Comment:
Optional:

I know it's not exported, so shouldn't be an issue, but maybe to be more accurate we could have here:
```suggestion
interface VersionField {
```
? 🤔

Reply:
I changed to `VersionField `: [f88c70a](https://github.com/cloudify-cosmo/cloudify-stage/pull/2247/commits/f88c70aa2f7279b67924993c28a022be67f4950f)